src/pages/Deposit.tsx

```diff
@@ -275,44 +305,47 @@ function Deposit({ poolName }: Props): ReactElement | null {
 function buildTransactionData(
   tokenFormState: TokensStateType,
   poolData: PoolDataType | null,
-  poolTokens: Token[],
-  poolLpToken: Token,
+  poolTokens: (BasicToken | undefined)[],
```

Comment:
Also I made a PR to introduce "ExpandedPools" which only contains pools that have full token data, so you don't have to do all `tokens?.[tokenAddr]` checking yourself 👍 
https://github.com/saddle-finance/saddle-frontend/pull/1193

Reply:
This needs to be revisited in the future. Null checks can't be done too high up in the component execution because of hook rules and if null checks are done below hooks then default values have to be added in child components eventually.